docs(load-balancing) update load-balancing algorithms (#2189)
Kong/docs.konghq.com: app/1.3.x/loadbalancing.md

### Balancing algorithms

The ring-balancer supports three load-balancing algorithms: `round-robin`,
`consistent-hashing`, and `least-connections`. By default, a ring-balancer
will use the `round-robin` algorithm.

> When using the `hash-based` algorithm, the input for the hash can be either

Author: There is not, good point! Fixed

`none`, `consumer`, `ip`, `header`, or `cookie`. When set to `none`, the
`round-robin` scheme will be used, and hashing will be disabled. The `hash-based`
algorithm supports a primary and a fallback hashing attribute; in case the primary
fails (e.g., if the primary is set to `consumer`, but no Consumer is authenticated),
the fallback attribute is used.

Supported hashing attributes are:

> - `none`: Do not use `hash-based`, but use `round-robin instead` (default).

Reviewer: ```suggestion
- `none`: Do not use `hash-based`; use `round-robin` instead (default).
```

> - `consumer`: Use the Consumer ID as the hash input. This option will fallback
>   on the Credential ID if no Consumer ID is available (in case of an external
>   authentication mechanism like LDAP).

Reviewer: ```suggestion
- `consumer`: Use the Consumer ID as the hash input. If no Consumer ID is available, it will fall back on the Credential ID (for example, in case of an external authentication mechanism like LDAP).
```

> - `ip`: The originating IP address will be used as input. Review the configuration

Reviewer: ```suggestion
- `ip`: Use the originating IP address as the hash input. Review the configuration
```

  settings for [determining the real IP][real-ip-config] when using this.

> - `header`: Use a specified header as input for the hash. The header name is

Reviewer: ```suggestion
- `header`: Use a specified header as the hash input. The header name is
```

  specified in either `hash_on_header` or `hash_fallback_header`, depending on whether
  `header` is a primary or fallback attribute, respectively.

> - `cookie`: Use a specified cookie with a specified path as input for the hash.

Reviewer: ```suggestion
- `cookie`: Use a specified cookie with a specified path as the hash input.
```

  The cookie name is specified in the `hash_on_cookie` field and the path is
  specified in the `hash_on_cookie_path` field. If the specified cookie is not
  present in the request, it will be set by the response. Hence, the `hash_fallback`
  setting is invalid if `cookie` is the primary hashing mechanism.

> The `consistent-hashing` algorithm is based on "Consistent Hashing" (or the

Reviewer: ```suggestion
The `consistent-hashing` algorithm is based on _Consistent Hashing_ (or the
```

> "Ketama Principle"), which makes sure that when the balancer gets modified by

Reviewer: ```suggestion
_Ketama Principle_), which ensures that when the balancer gets modified by
```

a change in its targets (adding, removing, failing, or changing weights), only
the minimum number of hashing losses occur. This maximizes upstream cache hits.

The ring-balancer also supports the `least-connections` algorithm, which selects

> the target with the least amount of connections, weighted by the Target's

Reviewer: ```suggestion
the target with the lowest number of connections, weighted by the Target's
```
